### DISCUSSION TITLE
Imgur upload plugin takes old ClientId for authentication

### DISCUSSION THREAD

**Discussion by ramananayak** [gavvvr/obsidian-imgur-plugin#39] [Getting help]
Hi, I was using authenticated imgur upload without any problems, but a week back it was showing issues , So I deleted the client id, revoked app access .Now I have regenerated clientId But every time now if I enter this new client ID and click on authenticate, I can see the URL still takes some other clientId and not the one which I entered . So I suspect, my old clientId is stored somewhere which I need to delete it. So any help would be appreciated on this. thanks !

**Comment by gavvvr** [comment]
Hi @ramananayak
You can not enter or use client id when you choose "authenticated upload". Client id is only used for anonymous upload:

https://github.com/gavvvr/obsidian-imgur-plugin/blob/e5260004a525b5249f77c6a39017c819ba65ce97/src/uploader/imgur/ImgurAnonymousUploader.ts#L17-L21

For authenticated upload, the plugin uses your OAuth token (which you silently receive by authenticating in your browser), client id is not used at all:
https://github.com/gavvvr/obsidian-imgur-plugin/blob/e5260004a525b5249f77c6a39017c819ba65ce97/src/imgur/ImgurClient.ts#L37-L41

I had some issues with authenticated upload recently too. I do not know the reason why. And I do not think there is anything I can do with it from plugin's source code, it's on imgur's side

**Reply by ramananayak** [reply]
I see ! thanks for clarification @gavvvr . So is it working fine for you now ? Just wanted to know so that I can try once again.

**Reply by gavvvr** [reply]
@ramananayak I did not make any changes to plugin's code. If it does not work sometimes, it's an Imgur's failure. The plugin's code is pretty straightforward

**Reply by ramananayak** [reply]
okay. thank you !
